### PR TITLE
Mosip 24268 the service type is incorrect in response body the description should be in logged in language

### PR DESCRIPTION
**Pull request by RitikJain4108**

### CODE REVIEW

File: resident/resident-service/src/main/java/io/mosip/resident/constant/ServiceType.java [hunk 1]
```diff
@@ -34,4 +34,14 @@ public static Optional<ServiceType> getServiceType(String serviceTypeString) {
         }
         return Optional.empty();
     }
+    
+    public static String getServiceTypeFromRequestType(RequestType requestType) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/constant/ServiceType.java]: Return Optional<String> and use Optional.empty() where null value is returned.

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java [hunk 2]
```diff
@@ -1836,16 +1835,16 @@ private List<String> convertServiceTypeToResidentTransactionType(String serviceT
 				if (service.equalsIgnoreCase(ServiceType.AUTHENTICATION_REQUEST.toString())) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java]: No need of if else condition, you can just pass the ServiceType obtained using `service` and pass it.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 3]
```diff
@@ -796,11 +796,11 @@ private void validateServiceType(String serviceType, String requestServiceHistor
 		if(serviceType!=null) {
 			List<String> serviceTypes = List.of(serviceType.split(","));
 			for (String service : serviceTypes) {
-				if (!service.equalsIgnoreCase(ResidentTransactionType.DATA_SHARE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.SERVICE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.ID_MANAGEMENT_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.DATA_UPDATE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.AUTHENTICATION_REQUEST.toString())
+				if (!service.equalsIgnoreCase(ServiceType.DATA_SHARE_REQUEST.name())
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: No need to check for each and every enum like this, create a single static method that will return getServiceTypeForStrig() where you will return Optional<ServiceType>.

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java [hunk 4]
```diff
@@ -796,11 +796,11 @@ private void validateServiceType(String serviceType, String requestServiceHistor
...
+						&& !service.equalsIgnoreCase(ServiceType.SERVICE_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.ID_MANAGEMENT_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.DATA_UPDATE_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.AUTHENTICATION_REQUEST.name())
 						&& !service.equalsIgnoreCase("ALL")) {
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: Add ALL also as a service type enum and map all request types to it, do it dynamically.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/constant/ServiceType.java (hunk 1 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/constant/ServiceType.java]: changes done

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java (hunk 2 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentServiceImpl.java]: changes done

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 3 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: changes done

File: resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java (hunk 4 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java]: changes done

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/IdAuthServiceImpl.java [hunk 5]
```diff
@@ -131,11 +133,11 @@ private void updateResidentTransaction(boolean verified,String transactionId, St
 			ResidentTransactionEntity residentTransaction = residentTransactionEntity.get(0);
 			if (residentTransaction != null) {
 				if(residentTransaction.getStatusCode().equalsIgnoreCase("OTP_REQUESTED")) {
-					residentTransaction.setRequestTypeCode(verified ? "OTP_VERIFIED" : "OTP_VERIFICATION_FAILED");
```
Comment (loganathan-sekaran) [inline resident/resident-service/src/main/java/io/mosip/resident/service/impl/IdAuthServiceImpl.java:134]: Preserve this Logic. This is needed for checking the OTP Verified status.

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/service/impl/IdAuthServiceImpl.java (hunk 5 above)
Comment (RitikJain4108) [reply resident/resident-service/src/main/java/io/mosip/resident/service/impl/IdAuthServiceImpl.java:134]: I changed only the request type...not the status code..

**Review by loganathan-sekaran — APPROVED** [review]

**Pull request merged**